### PR TITLE
LOVE 1034 - Make more basic aws/basic-eks-cluster

### PR DESCRIPTION
**Pull request by sffortytwo**
## Definition of Done

This is the definition of done for (new and modified) blueprints to be accepted into the [central XebiaLabs blueprints repository](https://github.com/xebialabs/blueprints).

### Value

This section describes how to determine whether a blueprint has enough value to be included in the central XebiaLabs blueprints repository.

- [x] The blueprint applies to a focused use case.
- [x] The blueprint uses at least one XebiaLabs product.
- [x] The blueprint depends on as few other products as possible. **N.B.:** This is part of the definition of done to keep the blueprint focused and applicable to as many users as possible. A blueprint that depends on one specific CI tool **and** one specific Java EE application server **and** one specific ticketing system can only be used by users that have all three products.
- [x] The blueprint can be used with content supplied by the user but also includes demo content. A blueprint that does not work with content supplied by the user is a demo or a showcase and does not help the user with their own work.

### Technical

This section describes how to determine whether the blueprint has the right technical quality to be included in the central XebiaLabs blueprints repository.

- [x] The branch from which the PR is created is up-to-date with the `development` branch.
- [x] The blueprint contains a `README.md` file at the root of the blueprint folder that describes the blueprint, using the README template in the `.github` folder of this repository.
- [x] The blueprint generates a `xebialabs/USAGE.md` file which, at a minimum, explains how to use the blueprint after it is generated (like adding any missing steps, creating accounts, setting up Docker containers, applying the YAML using `xl` CLI, running release. etc.). **N.B.:** Do not use this document to describe how to instantiate the blueprint. It will only be available to the user *after* the blueprint has been instantiated.
- [x] If the blueprint depends on external products that can be started as a Docker container, the blueprint generates a `docker/docker-compose.yml` file so that it can be tested without having to manually install those products. Do not use this Docker Compose file to start XL Deploy, XL Release, Docker or Kubernetes.
- [x] The blueprint does not contain sensitive information such passwords, tokens, credentials or licenses.
- [x] The blueprint uses `secret: true` in the `blueprint.yaml` parameter definition for question that ask for sensitive information.
- [x] The blueprint does not contain the files `xebialabs/.gitignore`, `xebialabs/values.xlvals` and `xebialabs/secrets.xlvals` and does not refer to them from the files section of the `blueprint.yaml` file. These files will be generated when the blueprint is instantiated. To generate the `xebialabs/values.xlvals` file, use the `saveInXlVals: true` directive in the parameters section of the blueprint. To generate the `xebialabs/secrets.xlvals` file, use the `secret: true` directive .
- [x] The blueprint does not define parameters for trivial things like phase names, task names, folder names etc. Ask questions that add value and be opinionated where possible. For example, ask for a project name and derive folder names, task names etc from that.
- [x] Folder and file names must use [kebab-case](http://wiki.c2.com/?KebabCase), for example: `aws/sample-app-demo`, `xld-environment.yaml`

### Review and testing

This section describes how to determine whether the blueprint has been reviewed and tested well enough to be included in the central XebiaLabs blueprints repository.

- [x] Unit test is added in a `__test__` folder for each blueprint. Refer the `CONTRIBUTING.md` file at the root of of this repository for more details.
- [x] The Travis CI for the PR is green
- [x] The blueprint has been reviewed by someone else in the team.
- [x] Both README and USAGE files have been reviewed by a technical writer and a product marketing manager.
- [x] The blueprint has been manually tested with the `docker/docker-compose.yml` that is generated as part of it.
- [x] The blueprint works on Linux, Mac and Windows.

### CODE REVIEW

File: aws/basic-eks-cluster/README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-# Amazon EKS Cluster
+# Basic AWS Kubernetes Engine Cluster
 
 ## Introduction
 
 Amazon Elastic Container Service for Kubernetes (EKS) allows you to deploy, manage, and scale containerized applications in the cloud using Kubernetes.
 
-Use this blueprint to provision a simple EKS cluster. The release template that the blueprint generates provisions a new cluster.
+Use this blueprint to provision a GKE cluster using CloudFormation.
```
Comment (deepu105) [inline aws/basic-eks-cluster/README.md]:
```suggestion
Use this blueprint to provision an EKS cluster using CloudFormation.
```

File: aws/basic-eks-cluster/README.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
-# Amazon EKS Cluster
+# Basic AWS Kubernetes Engine Cluster
```
Comment (deepu105) [inline aws/basic-eks-cluster/README.md]:
```suggestion
# Basic AWS Elastic Kubernetes Service(EKS) Cluster
```

File: aws/basic-eks-cluster/blueprint.yaml [hunk 3]
```diff
@@ -58,94 +55,23 @@ spec:
 
   - name: ClusterName
     type: Input
-    description: What is the name of the cluster?
+    description: What do you want to name the cluster?
 
-  - name: NodeAutoScalingGroupDesiredSize
```
Comment (deepu105) [inline aws/basic-eks-cluster/blueprint.yaml]: why are NodeAutoScalingGroupDesiredSize and NodeInstanceType removed. IMO they are needed even if the cluster is basic else this just becomes another hello world kind of thing. Else user would have to change those manually

File: aws/basic-eks-cluster/blueprint.yaml [hunk 4]
```diff
@@ -58,94 +55,23 @@ spec:
...
-  - path: xebialabs/xlr-pipeline-ci-cd.yaml.tmpl
-  - path: xebialabs/xlr-pipeline-destroy.yaml.tmpl
-  - path: xebialabs/USAGE.md.tmpl
+  - path: xebialabs/aws-basic-eks-cluster.yaml.tmpl
+  - path: xebialabs/aws-basic-eks-cluster-applications.yaml.tmpl
+  - path: xebialabs/aws-basic-eks-cluster-environments.yaml.tmpl
+  - path: xebialabs/USAGE-aws-basic-eks-cluster.md.tmpl
+
   - path: kubernetes/aws-auth-cm.yaml
   - path: cloudformation/cfn-secret-provider.zip
```
Comment (deepu105) [inline aws/basic-eks-cluster/blueprint.yaml]: I don't find this file in the PR?

File: aws/basic-eks-cluster/blueprint.yaml [hunk 5]
```diff
@@ -58,94 +55,23 @@ spec:
...
   - path: cloudformation/eks-master.yaml
   - path: cloudformation/eks-user.yaml
   - path: cloudformation/eks-vpc.yaml
   - path: cloudformation/eks-workers.yaml
-  - path: cloudformation/efs.yaml
-    dependsOnTrue: ProvisionEFS
-  - path: xebialabs.yaml
-  # docker-compose setup for required tools
```
Comment (deepu105) [inline aws/basic-eks-cluster/blueprint.yaml]: Please remove files as well

File: aws/basic-eks-cluster/blueprint.yaml (hunk 4 above)
Comment (sffortytwo) [reply aws/basic-eks-cluster/blueprint.yaml]: Do you mean `cloudformation/cfn-secret-provider.zip`. I definitely see it

File: aws/basic-eks-cluster/blueprint.yaml (hunk 3 above)
Comment (sffortytwo) [reply aws/basic-eks-cluster/blueprint.yaml]: I've added the scaling code back

File: aws/basic-eks-cluster/blueprint.yaml (hunk 5 above)
Comment (sffortytwo) [reply aws/basic-eks-cluster/blueprint.yaml]: File is removed

**Comment by sffortytwo** [comment]
I hear you. Scott and Daniel wanted a 'basic' AWS blueprint that behaved the same as the Azure and GCP ones. At least this is confined to the `v1-maintenance` branch and not `development`. I anyway want to investigate the possibility of using composability to add the XLR pipeline to the blueprints rather than excluding them with a Confirm.

**Comment by deepu105** [comment]
ok then

File: aws/basic-eks-cluster/blueprint.yaml (hunk 4 above)
Comment (deepu105) [reply aws/basic-eks-cluster/blueprint.yaml]: Yes, I see it only for the intermediate blueprint not for the basic one. If you search for it there is onely 1 file in the PR

**Pull request merged**